File: Open Kattis/tracksmoothing.py
import sys
import math

def dist(p0, p1):
    return math.sqrt((p0[0] - p1[0]) ** 2 + (p0[1] - p1[1]) ** 2)

def main():
    inp = [[int(num) for num in line.split()] for line in sys.stdin.read().splitlines()]
    N = inp[0][0]
    line = 1
    for i in range(N):
        R, V = inp[line]
        total_dist = 0
        x_0, y_0 = inp[line + 1]
        x0, y0, = x_0, y_0
        for i in range(line + 2,line + V + 1):
            x, y = inp[i]
            total_dist += dist((x0, y0), (x, y))
            x0, y0, = x, y

        total_dist += dist((x_0, y_0), (x, y))

        s = (total_dist - 2 * math.pi * R) / total_dist
        if s < 0:
            print("Not possible")
        else:
            print(s)

        line += V + 1

main()
        

    

File: Open Kattis/tracksmoothing.cpp
#include <iostream>
#include <cmath>
#define PI 3.14159265358979323846264338327950288419716939937510582

using namespace std;

double dist(int x0, int y0, int x1, int y1) {
	return sqrt((x0 - x1) * (x0 - x1) + (y0 - y1) * (y0 - y1));
}


int main() {
	int N;
	cin >> N;
	for (int i = 0; i < N; i++) {
		int R, V;
		
		double total_dist = 0;
		
		cin >> R >> V;
		int x_0, y_0;
		cin >> x_0 >> y_0;
		int x0 = x_0;
		int y0 = y_0;
		int x, y;
		for (int i = 0; i < V - 1; i++) {
			cin >> x >> y;
			total_dist += dist(x0, y0, x, y);
			x0 = x;
			y0 = y;
		}
		total_dist += dist(x_0, y_0, x, y);
		double s = (total_dist - 2 * PI * R) / total_dist;
		if (s < 0) {
			cout << "Not possible" << endl;

		} else {
			cout << s << endl;
		}
	}
	return 0;
}

		

		
		

